### PR TITLE
feat: support fit() in GeminiTextGenerator

### PR DESCRIPTION
**Pull request by ashleyxuu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes internal #343765747🦕

### CODE REVIEW

File: bigframes/ml/llm.py [hunk 1]
```diff
@@ -647,6 +651,55 @@ def _from_bq(
...
+        Returns:
+            GeminiTextGenerator: Fitted estimator.
+        """
+        if self._bqml_model.model_name.startswith("gemini-1.5"):
+            raise NotImplementedError("Fit is not supported for gemini-1.5 model.")
+
+        X, y = utils.convert_to_dataframe(X, y)
+
+        options = self._bqml_options
+        options["endpoint"] = "gemini-1.0-pro-002"
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:692]: Spill out the model version to be used  for tuning in docs.

File: bigframes/ml/llm.py [hunk 2]
```diff
@@ -647,6 +651,55 @@ def _from_bq(
...
+            X (bigframes.dataframe.DataFrame or bigframes.series.Series):
+                DataFrame of shape (n_samples, n_features). Training data.
+            y (bigframes.dataframe.DataFrame or bigframes.series.Series:
+                Training labels.
+
+        Returns:
+            GeminiTextGenerator: Fitted estimator.
+        """
+        if self._bqml_model.model_name.startswith("gemini-1.5"):
+            raise NotImplementedError("Fit is not supported for gemini-1.5 model.")
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:687]: Also mention supported model types in docs.

File: bigframes/ml/llm.py (hunk 1 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py:692]: Our model name is "gemini-pro-1.0", but their endpoint is called "gemini-1.0-pro-002". Any ideas for how to spill out it?

File: tests/system/load/test_llm.py [hunk 3]
```diff
@@ -99,3 +99,31 @@ def test_llm_palm_score_params(llm_fine_tune_df_default_index):
...
     assert all(col in score_result_col for col in expected_col)
+
+
+@pytest.mark.flaky(retries=2)
+def test_llm_gemini_configure_fit(llm_fine_tune_df_default_index, llm_remote_text_df):
+    model = bigframes.ml.llm.GeminiTextGenerator(
+        model_name="gemini-pro", max_iterations=1
+    )
+
+    df = llm_fine_tune_df_default_index.dropna().sample(n=100)
```
[the remark this answers is not in the document]
Comment (ashleyxuu) [reply tests/system/load/test_llm.py:110]: Thanks

File: bigframes/ml/llm.py (hunk 1 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py:692]: synced offline, no need to document the model version now: https://cloud.google.com/vertex-ai/generative-ai/docs/learn/model-versioning#auto-updated-models

File: bigframes/ml/llm.py (hunk 2 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py:687]: Done.

**Review by GarrettWu — APPROVED** [review]

**Pull request merged**